corehq/apps/custom_data_fields/templates/custom_data_fields/partials/fields_tab.html

```diff
@@ -7,169 +7,165 @@
+    </td>
+    <td>
+      <input class="form-control" type="text" data-bind="value: label, attr: { 'readonly': !isEditable() }"/>
+    </td>
+    <td>
+      <div class="form-check mt-2">
+        <input class="form-check-input" type="checkbox" data-bind="checked: is_required, attr: { 'disabled': !isEditable() }"/>
+      </div>
+    </td>
+    <td>
 
-        <div data-bind="visible: validationMode() === 'choice'">
-          <button data-bind="click: addChoice, attr: { 'disabled': !isEditable() }"
-                  type="button" class="btn btn-outline-primary btn-sm">
-            <i class="fa fa-plus"></i> {% trans "Add Choice" %}
+      {% if request|request_has_privilege:"REGEX_FIELD_VALIDATION" or couch_user.is_superuser %}
+        <div class="btn-group btn-group-separated validation-options">
+          <button data-bind="css: {active: !validationMode()},
+                          attr: { 'disabled': !isEditable() },
+                          click: function () {validationMode(undefined)}"
+               class="btn btn-outline-primary btn-sm">
+            {% trans "None" %}
+          </button>
+          <button data-bind="css: {active: validationMode() === 'choice'},
+                          attr: { 'disabled': !isEditable() },
+                          click: function () {validationMode('choice')}"
+               class="btn btn-outline-primary btn-sm">
+            {% trans "Choices" %}
           </button>
-          <ul data-bind="sortable: choices" class="list-unstyled validation-list">
-            <li data-bind="attr: {'data-order': _sortableOrder}">
-              <div class="input-group">
-                <span class="input-group-addon">  {# todo B5: css:input-group-addon #}
-                  <span data-bind="if: $parent.isEditable">
-                    <i class="fa-solid fa-up-down sortable-handle"></i>
-                  </span>
-                  <span data-bind="ifnot: $parent.isEditable">
-                    <i class="fa-solid fa-up-down"></i>
-                  </span>
+          <button data-bind="css: {active: validationMode() === 'regex'},
+                          attr: { 'disabled': !isEditable() },
+                          click: function () {validationMode('regex')}"
+               class="btn btn-outline-primary btn-sm">
+            {% trans "Regex" %}
+          </button>
+        </div>
+      {% endif %}
+
+      <div data-bind="visible: validationMode() === 'choice'">
+        <button data-bind="click: addChoice, attr: { 'disabled': !isEditable() }"
+                type="button" class="btn btn-outline-primary btn-sm">
+          <i class="fa fa-plus"></i> {% trans "Add Choice" %}
+        </button>
+        <ul data-bind="sortable: choices" class="list-unstyled validation-list">
+          <li data-bind="attr: {'data-order': _sortableOrder}">
+            <div class="input-group">
+              <span class="input-group">
```

Comment:
curious why the `input-group` is a span? And then also wrapped by an `input-group` parent `div`? I generally avoid `span`s wrapping anything other than text...

Reply:
unintentional, thanks - 790a57b276b